src/components/DataTable/DataTable.js

```diff
@@ -0,0 +1,364 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import Button from '../Button';
+import Icon from '../Icon';
+import Search from '../Search';
+
+export class DataTable extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      rows: props.initialRows,
+    };
+  }
+
+  render() {
+    const renderProps = {
+      rows: this.state.rows,
+    };
+
+    const { render, children } = this.props;
+
+    if (render !== undefined) {
+      return render(renderProps);
+    }
+
+    if (children !== undefined) {
+      return children(renderProps);
+    }
+
+    return null;
+  }
+}
+
+export class DataTableHeader extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      sorted: false,
+      active: false,
+    };
+  }
+
+  setActive = () => {
+    this.setState({
+      active: true,
+    });
+  };
+
+  removeActive = () => {
+    this.setState({
+      active: false,
+    });
+  };
+
+  handleClick = () => {
+    this.setState({
```

Comment:
Should be using callback method to avoid potential asynchronous issues.

```js
this.setState(({ sorted }) => {
  const nextSorted = !sorted;
  const direction = sorted ? 'desc' : 'asc'; // Maybe different order? Not sure what default sort is.
  this.props.onClick(this.props.sortBy, direction);
  return { sorted: !nextSorted };
});
```

Reply:
Most likely it'd be this variant, I think:

```js
this.setState(state => ({ sorted: !state.sorted }), () => {
   const direction = this.state.sorted ? 'desc' : 'asc';
   this.props.onClick(this.props.sortBy, direction);
});
```

Reply:
oooo, didn't even know setState could take more than one callback. Yes, that's better.

Reply:
Ah gotcha, will change it to that!